core/model/src/main/java/org/eclipse/sw360/antenna/model/artifact/facts/java/BundleCoordinates.java

```diff
@@ -62,6 +62,11 @@ public String getVersion() {
         return getBundleVersion();
     }
 
+    @Override
+    public String getPurl() {
+        return String.format("pkg:p2/%s@%s", getName(), getVersion());
```

Comment:
Instead of having this per coordinates implementation, can't we implement his in the base class and instead have the coordinates classes only implement a `getType()` method that is called by the base class implementation? At the example of this class, `getType()` should return "p2".

Reply:
just `getType` might not be enough: e.g. Docker has multiple ways of building a purl (with tag, with hash, ...)

Reply:
Right, still the `type` would be "docker" in all cases. `qualifiers` and `subpath` components could be added later without breaking the existing code.

Reply:
I changed that as you @sschuberth proposed. All coordinates we have now, are sufficiently encoded with the current functionality. So if we face more difficult Purl cases, the idea is to overwrite the getPurl method. Nonetheless, I like the idea that you can get a repository type from the coordinates.